lib/Github/Client.php

```diff
@@ -245,26 +293,56 @@ public function authenticate($tokenOrLogin, $password = null, $authMethod = null
      */
     public function setEnterpriseUrl($enterpriseUrl)
     {
-        $baseUrl = (substr($enterpriseUrl, -1) == '/') ? substr($enterpriseUrl, 0, -1) : $enterpriseUrl;
-        $this->getHttpClient()->client->setBaseUrl($baseUrl . '/api/v3');
+        $this->removePlugin(Plugin\AddHostPlugin::class);
+
+        $this->addPlugin(new Plugin\AddHostPlugin(UriFactoryDiscovery::find()->createUri($enterpriseUrl)));
+        $this->addPlugin(new PathPrepend(sprintf('/api/%s/', $this->getOption('api_version'))));
+    }
+
+    /**
+     * Add a new plugin to the chain
+     *
+     * @param Plugin $plugin
+     */
+    private function addPlugin(Plugin $plugin)
+    {
+        $this->plugins[] = $plugin;
+        $this->pluginsModified = true;
+    }
+
+    /**
+     * Remove a plugin by its fqn.
+     *
+     * @param string $fqn
+     */
+    private function removePlugin($fqn)
+    {
+        foreach ($this->plugins as $idx => $plugin) {
+            if ($plugin instanceof $fqn) {
+                unset($this->plugins[$idx]);
+            }
+        }
     }
 
     /**
-     * @return HttpClient
+     * @return HttpMethodsClient
      */
     public function getHttpClient()
     {
-        if (null === $this->httpClient) {
-            $this->httpClient = new HttpClient($this->options);
+        if ($this->pluginsModified) {
```

Comment:
you need to reset this property here, otherwise a new client will be created for each call to the getter